client/asset/eth/eth.go

```diff
@@ -0,0 +1,534 @@
+	}
+	eth.acctMtx.RLock()
+	defer eth.acctMtx.RUnlock()
+	var wallet rawWallet
+	findWallet := func() bool {
+		for _, w := range wallets {
+			for _, a := range w.Accounts {
+				if bytes.Equal(a.Address[:], eth.acct.Address[:]) {
+					wallet = w
+					return true
+				}
+			}
+		}
+		return false
+	}
+	if !findWallet() {
+		eth.log.Error("unable to find wallet for account: %v", eth.acct)
+		return false
+	}
+	return wallet.Status != "Unlocked"
+}
+
+// PayFee sends the dex registration fee. Transaction fees are in addition to
+// the registration fee, and the fee rate is taken from the DEX configuration.
+//
+// NOTE: PayFee is not intended to be used with Ethereum at this time.
+func (*ExchangeWallet) PayFee(address string, regFee uint64) (asset.Coin, error) {
+	return nil, notImplementedErr
+}
+
+// sendToAddr sends funds from acct to addr.
+func (eth *ExchangeWallet) sendToAddr(addr common.Address, amt, gasFee *big.Int) (common.Hash, error) {
+	tx := map[string]string{
+		"from":     fmt.Sprintf("0x%x", eth.acct.Address),
+		"to":       fmt.Sprintf("0x%x", addr),
+		"value":    fmt.Sprintf("0x%x", amt),
+		"gasPrice": fmt.Sprintf("0x%x", gasFee),
+	}
+	return eth.node.sendTransaction(eth.ctx, tx)
+}
+
+// Withdraw withdraws funds to the specified address. Fees are subtracted from
+// the value.
+//
+// TODO: Value could be larger than a uint64. Deal with it...
+// TODO: Return the asset.Coin.
+func (eth *ExchangeWallet) Withdraw(addr string, value uint64) (asset.Coin, error) {
+	eth.acctMtx.RLock()
+	defer eth.acctMtx.RUnlock()
+	_, err := eth.sendToAddr(common.HexToAddress(addr),
+		big.NewInt(0).SetUint64(value), big.NewInt(0).SetUint64(defaultGasFee))
+	if err != nil {
+		return nil, err
+	}
+	return nil, nil
+}
+
+// ValidateSecret checks that the secret satisfies the contract.
+func (*ExchangeWallet) ValidateSecret(secret, secretHash []byte) bool {
+	h := sha256.Sum256(secret)
+	return bytes.Equal(h[:], secretHash)
+}
+
+// Confirmations gets the number of confirmations for the specified coin ID.
+func (*ExchangeWallet) Confirmations(ctx context.Context, id dex.Bytes) (confs uint32, spent bool, err error) {
+	return 0, false, notImplementedErr
+}
+
+// SyncStatus is information about the blockchain sync status.
+func (eth *ExchangeWallet) SyncStatus() (bool, float32, error) {
+	// node.SyncProgress will return nil both before syncing has begun and
+	// after it has finished. In order to discern when syncing has begun,
+	// ensure we are connected to at least requiredNPeers, assume the node
+	// has started syncing from those peers were they ahead, and then defer
+	// to syncProgress.
+	peers, err := eth.node.peers(eth.ctx)
+	if err != nil {
+		return false, 0, err
+	}
+	if len(peers) < requiredNPeers {
+		return false, 0, nil
+	}
+	sp, err := eth.node.syncProgress(eth.ctx)
+	if err != nil {
+		return false, 0, err
+	}
+	if sp == nil {
+		return true, 1, nil
+	}
+	ratio := float32(sp.CurrentBlock) / float32(sp.HighestBlock)
```

Comment:
Wanna update syncstatus to match the recently merged server syncstatus approach?

Reply:
Yes.